src/multiselect/__tests__/multiselect.test.tsx

```diff
@@ -776,3 +778,17 @@ describe('Disabled item with reason', () => {
     expect(onChange).not.toHaveBeenCalled();
   });
 });
+
+test('group options can have label tag', () => {
+  const { wrapper } = renderMultiselect(<Multiselect options={groupOptions} selectedOptions={[]} />);
+  wrapper.openDropdown();
+
+  expect(wrapper.findDropdown().findOptionByValue('group')!.getElement().textContent).toBe('First categoryLabel tag');
+});
+
+test('group options can have tags', () => {
+  const { wrapper } = renderMultiselect(<Multiselect options={groupOptions} selectedOptions={[]} />);
+  wrapper.openDropdown();
+
+  expect(wrapper.findDropdown().findOptionByValue('group2')!.getElement().textContent).toBe('Second categoryTag1Tag2');
```

Comment:
Here you can use an existing `findTags` test-util method to check this expectation. Same for select test